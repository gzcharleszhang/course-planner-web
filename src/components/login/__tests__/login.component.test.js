import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Login from '../login.component'
import LoginContainer from '../login.container'

Enzyme.configure({ adapter: new Adapter() })

function setup() {
    const props = {
        loginUserAction: jest.fn()
    };

    const enzymeWrapper = shallow(<LoginContainer/>)

    return {
        props,
        enzymeWrapper
    }
}

describe('components', () => {
    describe('Header', () => {
        it('should render self and subcomponents', () => {
            const { enzymeWrapper } = setup();
            expect(enzymeWrapper.find('#email-field')).toBe(true);
            expect(enzymeWrapper.find('#password-field')).toBe(true);
            
            const loginProps = enzymeWrapper.find('Formik').props();
            expect(loginProps.newTodo).toBe(true);
        })

        // it('should call addTodo if length of text is greater than 0', () => {
        //     const { enzymeWrapper, props } = setup()
        //     const input = enzymeWrapper.find('TodoTextInput')
        //     input.props().onSave('')
        //     expect(props.addTodo.mock.calls.length).toBe(0)
        //     input.props().onSave('Use Redux')
        //     expect(props.addTodo.mock.calls.length).toBe(1)
        // })
    })
})