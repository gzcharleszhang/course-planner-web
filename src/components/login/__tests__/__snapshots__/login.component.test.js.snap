// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render the component 1`] = `
<ContextProvider
  value={
    Object {
      "dirty": false,
      "errors": Object {},
      "handleBlur": [Function],
      "handleChange": [Function],
      "handleReset": [Function],
      "handleSubmit": [Function],
      "initialValues": Object {
        "email": "",
        "password": "",
      },
      "isSubmitting": false,
      "isValid": false,
      "isValidating": false,
      "registerField": [Function],
      "resetForm": [Function],
      "setError": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFormikState": [Function],
      "setStatus": [Function],
      "setSubmitting": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "status": undefined,
      "submitCount": 0,
      "submitForm": [Function],
      "touched": Object {},
      "unregisterField": [Function],
      "validate": undefined,
      "validateField": [Function],
      "validateForm": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
      "validationSchema": ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "password",
          "email",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "email": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "undefined": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "password": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      },
      "values": Object {
        "email": "",
        "password": "",
      },
    }
  }
>
  <Form>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <FormikConnect(FieldInner)
        className="form-control"
        name="email"
        type="text"
      />
      <FormikConnect(ErrorMessageImpl)
        className="invalid-feedback"
        component="div"
        name="email"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <FormikConnect(FieldInner)
        className="form-control"
        name="password"
        type="password"
      />
      <FormikConnect(ErrorMessageImpl)
        className="invalid-feedback"
        component="div"
        name="password"
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="btn btn-primary mr-2"
        type="submit"
      >
        Login
      </button>
      <button
        className="btn btn-secondary"
        type="button"
      >
        Signup
      </button>
    </div>
  </Form>
</ContextProvider>
`;
